### PR TITLE
Create python lint action

### PR DESCRIPTION
**Pull request by thrile**

### CODE REVIEW

**Comment by thrile** [comment]
I personally love these types of static checks for silly things I miss in my code. I don't know if we need to make changes to the action or to the code to get it to pass?

File: .github/workflows/pylint.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.7
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.7
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install pylint
```
Comment (thrile) [inline .github/workflows/pylint.yml:19]: @jameswolfeoliver does this make sense?

File: .github/workflows/pylint.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    - name: Analysing the code with pylint
+      run: |
+        pylint `ls -R|grep .py$|xargs`
```
Comment (thrile) [inline .github/workflows/pylint.yml:22]: open to adding any pylint options or changing how it picks files

**Review by jameswolfeoliver — APPROVED** [review]
Looks good

I added a pylintrc file so we can disable some checks and ignore the reference code

**Review by jameswolfeoliver — CHANGES_REQUESTED** [review]
Hollup, some weird stuff happened with pip

**Review by jameswolfeoliver — APPROVED** [review]

**Pull request merged**